drivers/interrupt_controller/intc_plic.c

```diff
@@ -146,17 +190,21 @@ static int riscv_plic_is_edge_irq(const struct device *dev, uint32_t local_irq)
 static void plic_irq_enable_set_state(uint32_t irq, bool enable)
 {
 	const struct device *dev = get_plic_dev_from_irq(irq);
-	const struct plic_config *config = dev->config;
 	const uint32_t local_irq = irq_from_level_2(irq);
-	mem_addr_t en_addr = config->irq_en + local_irq_to_reg_offset(local_irq);
-	uint32_t en_value;
-	uint32_t key;
 
-	key = irq_lock();
-	en_value = sys_read32(en_addr);
-	WRITE_BIT(en_value, local_irq & PLIC_REG_MASK, enable);
-	sys_write32(en_value, en_addr);
-	irq_unlock(key);
+	for (uint16_t i = 0; i < arch_num_cpus(); i++) {
```

Comment:
`arch_num_cpus()` has `uint32_t` type

```suggestion
	for (uint32_t i = 0; i < arch_num_cpus(); i++) {
```

nit: `i` can be renamed to `cpu_num` as well for consistency with `plic_init`